source/ui/Components/editor/IssueMark.qml

```diff
@@ -0,0 +1,234 @@
+
+                // The issue message to display.
+                property var issueMessage: "";
+                // The type of issue (error, warning, information).
+                property var issueType: "";
+                // Margin between issue item icon, issue text and right edge.
+                property var _textMargin: 4
+
+                color: {
+                    switch (issueType) {
+                    case "Error":
+                        return errorColorSolid;
+                    case "Warning":
+                        return warningColorSolid;
+                    case "Information":
+                        return informationColorSolid;
+                    default:
+                        return errorColorSolid;
+                    }
+                }
+
+                // TextMetrics do not take line breaks into account for their width. Therefore the
+                // issueMarkTextBackground should be just as wide as the issueTextMetrics' width but should
+                // always fit into the editor (
+                width: Math.min(issueTextMetrics.width, issueText.width) + 3 * _textMargin + issueItemIcon.width
+                height: Math.max(issueText.height, issueLineHighlight.height)
+
+                // Each issueItem displays an icon corresponding to its issueType.
+                Image {
+                    id: issueItemIcon
+                    width: issueIcon.width
+                    height: issueIcon.height
+                    anchors.left: parent.left
+                    anchors.leftMargin: _textMargin
+                    anchors.verticalCenter: parent.verticalCenter
+                    source: {
+                        switch (issueType) {
+                        case "Error":
+                            return "Issue Icons/Error Icon.png";
+                        case "Warning":
+                            return "Issue Icons/Warning Icon.png";
+                        case "Information":
+                            return "Issue Icons/Information Icon.png";
+                        default:
+                            return "Issue Icons/Error Icon.png";
+                        }
+                    }
+                }
+
+                // Required for finding out whether issue text overlaps line text.
+                TextMetrics {
+                    id: issueTextMetrics
+                    text: issueText.text
+                    font: issueText.font
+                }
+
+                // Displays the issueMessage (error message, warning message, information message).
+                Text {
+                    id: issueText
+                    anchors.right: parent.right
+                    anchors.rightMargin: _textMargin
+                    anchors.verticalCenter: parent.verticalCenter
+                    wrapMode: Text.WrapAtWordBoundaryOrAnywhere
+                    width: scrollView.width - sidebar.width - issueItemIcon.width - 3*_textMargin

```

Comment:
I don't know exactly what the problem is, but the the error text width doesn't properly update when zooming the editor. Look at the editor components for scaling, I think replacing `scrollView.width` by [this](https://github.com/TUM-LRR/era-gp-sim/blob/ec6ba63d8ca9f5eece8444551bd8d8ed817daffb/source/ui/Components/editor/Editor.qml#L154) might help. But I have a feeling something else has to be changed, too 😉